api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -336,6 +342,10 @@ func (m *AzureManagedControlPlane) validateVersion(_ client.Client) error {
 		return errors.New("must be a valid semantic version")
 	}
 
+	if _, err := semverv4.ParseTolerant(m.Spec.Version); err != nil {
```

Comment:
Have you thought about using [this package](https://pkg.go.dev/golang.org/x/mod/semver#IsValid) instead? It seems like it has a similar function for [compare](https://pkg.go.dev/golang.org/x/mod/semver#Compare) that you're looking for.

Reply:
ack